src/VisualStudio/Core/Def/Telemetry/AggregatingTelemetryLog.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.VisualStudio.Telemetry;
+using Microsoft.VisualStudio.Telemetry.Metrics;
+using Microsoft.VisualStudio.Telemetry.Metrics.Events;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Telemetry
+{
+    /// <summary>
+    /// Provides a wrapper around the VSTelemetry histogram APIs to support aggregated telemetry. Each instance
+    /// of this class corresponds to a specific FunctionId operation and can support aggregated values for each
+    /// metric name logged.
+    /// </summary>
+    internal sealed class AggregatingTelemetryLog : ITelemetryLog
+    {
+        private readonly IMeter _meter;
+        private readonly TelemetrySession _session;
+        private readonly Dictionary<string, IHistogram<int>> _histograms;
+        private readonly object _lock;
+        private readonly HistogramConfiguration? _histogramConfiguration;
+        private readonly string _eventName;
+
+        private const string MeterVersion = "0.39";
+
+        public AggregatingTelemetryLog(TelemetrySession session, FunctionId functionId, double[]? bucketBoundaries)
```

Comment:
when do we not have boundaries?

Reply:
Added doc comments in commit 7 for this method describing what this parameter is. None of the current callers end up specifying bucket boundaries, which means we pass in a null configuration to the CreateHistogram call and use the default bucket boundaries defined in vs telemetry.

Reply:
ok.  if none of the current callers specify, shiould we just remove this for now?  we could add in later i f needed.

Reply:
i'm also curiosu what the default buckets are.  do you think they give enough granularity above hte 500ms time?